proposal.md

```diff
@@ -32,6 +32,29 @@ By dedicating time specifically to maintenance we hope to reduce the open issue
 backlog and increase the growth rate of new contributors, as pandas becomes an
 easier project to contribute to.
 
+### Extension Types
+
+Pandas Extension Types allow for extending NumPy types with custom
+data types and array storage. Pandas uses extension types internally for several
+type such as categorical, datetime with timezone and interval data, and provides
+an interface for 3rd-party libraries to define their own custom data types.
+
+Many parts of pandas still don't full handle such extension types, leading to
+special cases in the code (hurting maintainability), places where the data
+is unintentionally converted to a NumPy array (with possible performance
+implications) and missing functionalities for those data types.
+These problems are especially pronounced for nested data.
```

Comment:
additional one is missing support for extension types in the Index in general, should we mention that explicitly?

Reply:
I'll add a sentence on it.